### DISCUSSION TITLE
Test results with Project Activate

### DISCUSSION THREAD

**Discussion by NachumLehman** [ATMakersOrg/EyeAlert#14] [General]
I had someone in my office spend the entire day yesterday trying all Tobii devices with Project Activate to see how it needed to be positioned next to an eye tracker for optimal performance and what that felt like.  All of the models including the Tobii i16 which is the widest one allowed the Android to be mounted on the side and Project Activate was able to accurately detect an eye movement to the left. There were no false positives while he was using the eye tracker even at the leftmost areas.  

It was important for me to see this work by us before sending it out to pALS.  

![Tobii test with Project Activate](https://user-images.githubusercontent.com/104594850/166706285-d5e4a012-592a-4dbb-9dee-06d1e7d7c673.jpg)
.

**Comment by ImproveAbility** [comment]
Thanks for the testing!  I am having two clients try it out while we are with them to see what position and movements work for them.  I will report back.

**Comment by nelsonii** [comment]
That's very promising news. I love that we'll have multiple solutions to the problem -- every user is different, it seems.

**Comment by ImproveAbility** [comment]
Follow up notes from call with Lisle.  Out the box Project Activate is set less sensitive.  Most users will benefit from adjusting sensitivity of the motion that they want to use.  She said that phones are not designed to have their cameras on all the time, so it is a good practice to reboot phone on a regular basis.  Best use would be if someone needs it a certain time of the day - not 24/7 as phones just don't like that.  It should also be noted that Project Activate was a one time project for Google and is not longer being updated.  Lisle did say that if she were able to get an intern or other interested party to put some time into it, she would love to get it open source so that it could be built upon and updated.  Google did their testing on Pixel 2 and Pixel 3 phones.

**Comment by JayToltTech** [comment]
FYI, I reported the following bug to Lisie from running Project Activate on my Samsung Galaxy S10 phone:

bash-3.2$ adb logcat -v time AndroidRuntime:E *:S
--------- beginning of crash
07-28 11:14:41.634 E/AndroidRuntime(27253): FATAL EXCEPTION: main
07-28 11:14:41.634 E/AndroidRuntime(27253): Process: com.google.android.apps.vision.switches, PID: 27253
07-28 11:14:41.634 E/AndroidRuntime(27253): java.lang.IllegalStateException: withStartedExpression must be called from ExpressionState READY, NEUTRAL or COMPLETED, not CANCELLED
07-28 11:14:41.634 E/AndroidRuntime(27253): at com.google.common.base.Preconditions.checkState(Preconditions.java:601)

This is a basic "oops" bug in the UI, will trigger somewhat randomly (I couldn't find a reliable reproduction case) and cause an app crash, and it's not related to the hardware of the phone, it's just your standard bug in how the app works.

**Reply by JayToltTech** [reply]
To capture this I:

* Installed Android Studio on my MacBook (https://developer.android.com/studio)
* Set my Samsung phone into Developer Mode (https://developer.android.com/studio/debug/dev-options)
* Went into Developer Settings on the phone and turned on USB Debugging
* Attached the phone to my Macbook with a USB cable
* Opened Terminal on my Macbook
* Ran the adb logcat command above

**Reply by JayToltTech** [reply]
You can run this command long after the crash has occurred and it will go find the old crash logs.

**Comment by JayToltTech** [comment]
Note that Project Activate doesn't have the ability to trigger Punt37 (unlike the earlier prototype) :-(. So in addition to fixing the bug(s), it would need to be updated to support Settings -> Shortcut -> Your device will... -> Call Punt37 to restart your SGD

**Comment by JayToltTech** [comment]
FYI, I ran a 4 day test.  My test environment was:

* Samsung Galaxy S10 (2019 flagship Android phone)
* Pixel Desk Stand (with QI power charger)
* [Caffeine App](https://play.google.com/store/apps/details?id=moe.zhs.caffeine&gl=US) to keep the screen on and phone active 24x7
* [Project Activate](https://play.google.com/store/apps/details?id=com.google.android.apps.vision.switches&gl=US) v13.0 published Sept 19, 2021

The app crashed three times for me, all in the same line of code (see above, ExpressionState.CANCELLED), each time when it was transitioning from one state to another.

The app was not killed by the OS.  The app seemed to have reasonable long term performance, it did not get sluggish or slow, assuming it didn't crash from the above bug.

@NachumLehman @ImproveAbility

**Comment by ImproveAbility** [comment]
Thanks for your feedback Jay - I appreciate the time you spent on this.  I think that after our talk with Leise and our testing, this is not a viable option for what we are looking for.  I know it feels like this project has stalled, but please know that it has not.  I still have some testing to do on the IR watchdog devices.  My understanding is that we are at a bit of a standstill until Rasberry Pis are available again.  But either way, I am thinking we need another pow-wow session.  I will try to get something on the calendar.  My schedule is a bit blocked at the moment as we are in the thick of the end of the state's fiscal year, and I am training a new rep.  So, I may not be able to come up for air until September.  But, please know that I have not forgotten about this project.

**Comment by NachumLehman** [comment]
YES Thank you Jay. This is extremely valuable information.  I hope Lisie
will be able to get that bug addressed. The IR watchdog seems to be working
well. It needs some minor tweaking. Antoinette, the watchdog does
not use the Raspberry Pi's so we should be able to move on that one
quickly.  They need the Raspberry Pi for the eye gesture call bell.

Thanks all!
Nachum

On Mon, Aug 1, 2022 at 11:57 AM Antoinette Verdone ***@***.***>
wrote:

> Thanks for your feedback Jay - I appreciate the time you spent on this. I
> think that after our talk with Leise and our testing, this is not a viable
> option for what we are looking for. I know it feels like this project has
> stalled, but please know that it has not. I still have some testing to do
> on the IR watchdog devices. My understanding is that we are at a bit of a
> standstill until Rasberry Pis are available again. But either way, I am
> thinking we need another pow-wow session. I will try to get something on
> the calendar. My schedule is a bit blocked at the moment as we are in the
> thick of the end of the state's fiscal year, and I am training a new rep.
> So, I may not be able to come up for air until September. But, please know
> that I have not forgotten about this project.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ATMakersOrg/EyeAlert/discussions/14#discussioncomment-3300098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY573IRIUAMLXVFRHYGIKO3VW7XV5ANCNFSM5VCHJRAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

*Nachum Lehman*
*Technical Director*
*Office * 718-313-0030
*Direct * 718- <718-874-0040>644-1753
www.bridgingvoice.org

**Reply by ImproveAbility** [reply]
Nachum, yes, I understand that Rasberry Pi not needed for IR watchdog, I am sorry if I confused anyone.  I was just thinking about all of the solutions we discussed for this problem.
